sklearn/neighbors/_classification.py

```diff
@@ -135,8 +135,7 @@ class KNeighborsClassifier(KNeighborsMixin, ClassifierMixin, NeighborsBase):
 
        Regarding the Nearest Neighbors algorithms, if it is found that two
        neighbors, neighbor `k+1` and `k`, have identical distances
-       but different labels, the results will depend on the ordering of the
-       training data.
+       but different labels, the smallest label will be returned.
```

Comment:
In the current algorithm, it's based on the order of the training data and not the smallest label.

In the following example, `KNeighborsClassifier` is always using the data point that appears first in the training data:

```python
from sklearn.neighbors import KNeighborsClassifier
import numpy as np

X1 = np.asarray([
    [1, 0],
    [0, 1],
])
y1 = np.asarray([0, 1])
nn = KNeighborsClassifier(n_neighbors=1).fit(X1, y1)

print(nn.predict([[0, 0]]))
# [0]

# Change the order of the data points
X2 = np.asarray([
    [0, 1],
    [1, 0],
])
y2 = np.asarray([1, 0])
nn = KNeighborsClassifier(n_neighbors=1).fit(X2, y2)

print(nn.predict([[0, 0]]))
# [1]
```